server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -278,38 +288,43 @@ private void setEnabled(boolean enabled) {
     private void setLowWatermark(String lowWatermark) {
         // Watermark is expressed in terms of used data, but we need "free" data watermark
         this.lowWatermarkRaw = lowWatermark;
-        this.freeDiskThresholdLow = 100.0 - thresholdPercentageFromWatermark(lowWatermark);
-        this.freeBytesThresholdLow = thresholdBytesFromWatermark(
+        this.freeDiskThresholdLow = 100.0 - DiskThresholdSettingParser.parseThresholdPercentage(lowWatermark);
+        this.freeBytesThresholdLow = DiskThresholdSettingParser.parseThresholdBytes(
             lowWatermark,
             CLUSTER_ROUTING_ALLOCATION_LOW_DISK_WATERMARK_SETTING.getKey()
         );
+        notifyListeners();
```

Comment:
I don't think we need another observer mechanism in place as these methods are already listeners to settings update https://github.com/elastic/elasticsearch/pull/88175/files#diff-697233b32fbd24830c548746773a3013fc03a1cacc6ca877c8d75f0b9f68f751L121

Could the HealthMetadataService be a listener on the settings and consume the "settings changed" notifications that way?

Reply:
Fixed